### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by joeglDev**
Ticket 4: GET /api/articles/:article_id

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -33,6 +32,54 @@ describe(Endpoints.ALL_TOPICS_END, () => {
   });
 });
 
+//tests for GET /api/articles/:article_id
+describe(Endpoints.ARTICLE_BY_ID_END, () => {
+  test("GET request returns a status code of 200", () => {
+    return request(app).get("/api/articles/3").expect(200);
+  });
+  test("GET request returns an article object on key of article", () => {
```
Comment (alexdevdavis) [inline __tests__/endpoints.test.js:40]: try to be consistent with the format of your code:
e.g. GET returns ____

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -33,6 +32,54 @@ describe(Endpoints.ALL_TOPICS_END, () => {
...
+      .then(({ body: err }) => {
+        expect(err.msg).toBe(
+          "Error 404: No articles found for article_id 6767."
+        );
+      });
+  });
+  test("Get request with invalid id returns http status 400", () => {
+    return request(app).get("/api/articles/notNumber").expect(400);
+  });
+  test("Get request invalid id returns an error object", () => {
```
Comment (alexdevdavis) [inline __tests__/endpoints.test.js:73]: For efficiency, consider combining the status and expected; something like '400: returns an error when passed invalid article id'

File: errors.js [hunk 3]
```diff
@@ -1,4 +1,21 @@
 module.exports.handleInvalidPaths = (req, res, next) => {
-    const responseBody = {status : 400, msg : "Error 400: Path invalid."};
+  const responseBody = { status: 400, msg: "Error 400: Path invalid." };
+  res.status(400).send(responseBody);
+};
+
+module.exports.handleCustomErrors = (err, req, res, next) => {
+  if (err.status && err.msg) {
```
Comment (alexdevdavis) [inline errors.js:7]: think about how this condition might cause some bugs down the line

**Review by alexdevdavis — APPROVED** [review]
Your code is good. The comments I've added are points for your consideration, but I'm happy for you to merge this code. Nice one

File: __tests__/endpoints.test.js (hunk 1 above)
Comment (joeglDev) [reply __tests__/endpoints.test.js:40]: will do going forward - is also a possible issue with naming commits

File: __tests__/endpoints.test.js (hunk 2 above)
Comment (joeglDev) [reply __tests__/endpoints.test.js:73]: ok

**Pull request merged**